src/main/java/org/folio/service/orders/OrderLinesSummaryPopulateService.java

```diff
@@ -56,31 +61,48 @@ public Future<CompositeOrderRetrieveHolder> populate(CompositeOrderRetrieveHolde
   public Future<Double> calculateTotalEstimatedPrice(List<CompositePoLine> compositePoLines,
       RequestContext requestContext) {
     return configurationEntriesCache.getSystemCurrency(requestContext)
-      .compose(toCurrency -> getCollect(compositePoLines, requestContext, toCurrency)
-        .map(amounts -> amounts.stream()
-        .reduce(Money.of(0, toCurrency), Money::add)
-        .getNumber()
-        .doubleValue()));
+      .compose(toCurrency -> getExchangeRatesPerPoLine(compositePoLines, toCurrency, requestContext)
+      .compose(poLineExchangeRate -> Future.succeededFuture(Pair.of(toCurrency, poLineExchangeRate)))
+      .compose(toCurrencyPolExcRates -> Future.succeededFuture(convertEstimatedPrice(compositePoLines, toCurrencyPolExcRates))));
```

Comment:
with .map would be simpler here
```suggestion
.map(poLineExchangeRate -> Pair.of(toCurrency, poLineExchangeRate))
.map(toCurrencyPolExcRates -> convertEstimatedPrice(compositePoLines, toCurrencyPolExcRates)));
```